server/src/com/treode/server/SchematicStore.scala

```diff
@@ -22,6 +26,43 @@ class SchematicStore (store: Store, schema: Schema) {
     store.write (tx, ct, ops:_*)
   }
 
+  def batch_write(tx: TxId, ct: TxClock, req: Request): Async [TxClock] = {
+    val mapper = new ObjectMapper
+    val node = mapper.readTree(req.getContentString)
+    val it = node.iterator
+    var ops = Seq [WriteOp] ()
+    val pairs = new HashSet [(String, String)]
+    
+    for (request <- it) {
+      val table = request.get("table") .textValue
+      val key = request.get("key") .textValue
+      val op = request.get("op") .textValue
+      if (pairs.contains((table, key))) {
+        throw new DuplicatePairsException
+      } else {
+        pairs += ((table, key))
+        op match {
```

Comment:
Make this match case insensitive. Do `op.toLowerCase match`. And then lower case the strings below of course.